### ISSUE TITLE
Event.MouseWheel.Delta sometimes returns 0

### ISSUE THREAD

**Issue by LaurentGomila** [LaurentGomila/SFML#95]
On Windows, when a mouse wheel event is triggered, sometimes the Delta parameter is 0. It probably happens with high-resolution mice, which trigger mouse wheel events with an internal delta less than 120. Therefore this code:

    event.MouseWheel.Delta = static_cast<Int16>(HIWORD(wParam)) / 120;

... leads to Delta being 0.

I must find a solution to handle correctly both regular and high-resolution mice.

**Comment by Addisonbean** [comment]
Is there a workaround for this in the meantime?

**Comment by binary1248** [comment]
> Is there a workaround for this in the meantime?

Not really...

So... floats it is then? :grin:

**Comment by eXpl0it3r** [comment]
Sounds good to me.

**Comment by LaurentGomila** [comment]
This would be an API breaking change, right?

**Comment by binary1248** [comment]
Since floats can be cast back to integers (with warnings if you have them enabled) I don't see how this would "break" the API. Strictly speaking, yes it would break the API but the same code could be compiled even if floats were used, so in that sense it doesn't really "break" anything for the developer. The ABI would change yes... but that would happen no matter how we choose to solve this problem I think.

Bonus: If the developer already uses C++11 automatic type deduction and has type independent code, they won't even notice anything.

**Comment by zsbzsb** [comment]
>This would be an API breaking change, right?

I don't see how this would be any different than changing font metrics from ints to floats.

**Comment by LaurentGomila** [comment]
```cpp
if (event.mouseWheel.delta == 1)
    up();
else if (event.mouseWheel.delta == -1)
    down();
```
Not the most clever code, but you get the idea, code that deals with this event may do something that works with integers but not floats (in this case, ```event.mouseWheel.delta``` might be 1.00001).

**Comment by binary1248** [comment]
Well... we could always consider just adding a .deltaHighPrecision field to the struct instead.

**Comment by MarioLiebisch** [comment]
> Not the most clever code, but you get the idea, code that deals with this event may do something that works with integers but not floats (in this case, event.mouseWheel.delta might be 1.00001).

I'd probably blame the compiler since it's not warning about the potential issue. :)

But considering this would be very isolated and pretty easy to locate/fix, I'd ignore that. As mentioned in the other thread, keeping the integer comparison will be buggy as well. So do we want a consistent and clean fix or do we want to keep a broken system just to lessen the impact on existing code, which would essentially be broken anyway?

**Comment by binary1248** [comment]
We provide a clean way of doing things _for those who care_. There are always going to be people who can't care less about their broken code/warnings, and for them the current implementation is just "good enough". They will start caring however, if we break (i.e. no longer function "good enough") their code. I don't think annoying people who don't want to be annoyed is the right way to go. For those who do care and commented on this issue, there will be the new system with which they can write their code and benefit from.

The API isn't meant to teach people how to write their code. If they can't be bothered using it properly and are fine with the results thereof, then so be it.

**Comment by LaurentGomila** [comment]
Not breaking the API between minor versions, especially with such nasty bugs that may end up being hard to track for users, is *very important* if we want SFML to be considered a little more seriously. Users who update from 2.2 to 2.3 expect their old code to continue to work as expected, not to have to track stupid bugs that the dev introduced.

**Comment by MarioLiebisch** [comment]
Yeah, well, agree - just no fan of having "duplicate" members.

**Comment by eXpl0it3r** [comment]
Fixed in #810
